### PR TITLE
Add filter non user option

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs [hunk 1]
```diff
@@ -130,5 +133,32 @@ private string ParsePath(string line)
...
+                var response = client.GetAsync(userUriStub);
+                if (response.Result.IsSuccessStatusCode)
+                {
+                    return false;
+                }
+                return true;
+            }
+            catch
+            {
+                Console.WriteLine($"Http call {userUriStub} to failed.");
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:159]: Let's not write to the console.

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs [hunk 2]
```diff
@@ -130,5 +133,32 @@ private string ParsePath(string line)
...
+        {
+            client.BaseAddress = new Uri("https://api.github.com/");
+            client.DefaultRequestHeaders.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("CodeOwnerRetriever", "1.0"));
+            client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+            Owners.RemoveAll(r => IsTeamAlias(r));
+        }
+
+        private static bool IsTeamAlias(string alias)
+        {
+            var userUriStub = $"users/{alias}";
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:147]: Given we are filtering to users as opposed to directly checking teams we should change the naming to IsUserAlias

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs [hunk 3]
```diff
@@ -130,5 +133,32 @@ private string ParsePath(string line)
...
+
+        public void FilterOutTeamAlias()
+        {
+            client.BaseAddress = new Uri("https://api.github.com/");
+            client.DefaultRequestHeaders.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("CodeOwnerRetriever", "1.0"));
+            client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+            Owners.RemoveAll(r => IsTeamAlias(r));
+        }
+
+        private static bool IsTeamAlias(string alias)
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:145]:
```suggestion
        private static bool IsGitHubUserAlias(string alias)
```

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs [hunk 4]
```diff
@@ -130,5 +133,32 @@ private string ParsePath(string line)
             // remove the path from the string.
             return line.Substring(ownerStartPosition);
         }
+
+        public void FilterOutTeamAlias()
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:137]:
```suggestion
        public void FilterOutNonUserAlias()
```

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners/Program.cs [hunk 5]
```diff
@@ -11,11 +11,13 @@ class Program
...
         /// <param name="codeOwnerFilePath">The path of CODEOWNERS file in repo</param>
         /// <param name="targetDirectory">The directory whose information is to be retrieved</param>
+        /// <param name="filterTeamAlias">The option to filter out code owner team alias.</param>
         /// <returns>Exit code</returns>
 
         public static int Main(
             string codeOwnerFilePath,
-            string targetDirectory
+            string targetDirectory,
+            bool filterTeamAlias = false
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners/Program.cs:20]:
```suggestion
            bool filterOutNonUserAliases = false
```

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs (hunk 1 above)
Comment (sima-zhu) [reply tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs:159]: Do we want to throw http native error message?

Comment (sima-zhu) [reply tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs]: This is error flow. Try to print custom error message. If removed, then we do not need to have try block. Do you mean this?

**Review by weshaggard — APPROVED** [review]

File: tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs (hunk 1 above)
Comment (weshaggard) [reply tools/code-owners-parser/CodeOwnersParser/CodeOwnerEntry.cs]: Yes just let the exception flow for now. We might need to consider retries later if we find this isn't reliable enough.

**Review by weshaggard — APPROVED** [review]
Please test this with and without a user account to make sure the exception flow doesn't break things.

**Comment by sima-zhu** [comment]
Done with the testing on cpp template.
Command:  
```
"commandLineArgs": "--target-directory \"/sdk/template\" --code-owner-file-path \"C:\\sdk\\azure-sdk-for-cpp\\.github\\CODEOWNERS\" --filter-out-non-user-aliases "
```
image

Command: 
```
"commandLineArgs": "--target-directory \"/sdk/template\" --code-owner-file-path \"C:\\sdk\\azure-sdk-for-cpp\\.github\\CODEOWNERS\" ",
```
image

**Pull request merged**